addons/mass_mailing_sms/wizard/mailing_sms_test.py

```diff
@@ -24,16 +24,44 @@ def action_send_sms(self):
         invalid_numbers = [number for number, info in sanitize_res.items() if info['code']]
         if invalid_numbers:
             raise exceptions.UserError(_('Following numbers are not correctly encoded: %s, example : "+32 495 85 85 77, +33 545 55 55 55"', repr(invalid_numbers)))
-        
+
         record = self.env[self.mailing_id.mailing_model_real].search([], limit=1)
         body = self.mailing_id.body_plaintext
         if record:
             # Returns a proper error if there is a syntax error with jinja
             body = self.env['mail.render.mixin']._render_template(body, self.mailing_id.mailing_model_real, record.ids)[record.id]
 
-        self.env['sms.api']._send_sms_batch([{
-            'res_id': 0,
+        sent_sms_list = self.env['sms.api']._send_sms_batch([{
+            'res_id': number,  # used to map the result to the number to log notifications
```

Comment:
Mmmh ... that's not the purpose of this field.

Reply:
On their docstring it's marked as an "ID of a client side record".
So I figured it would ease the code here instead of creating a "dummy IDs to numbers" map or trusting the order which is not really guaranteed since it's an external API.

I don't see any real harm to it but I can change if it triggers you.

Reply:
Putting the number is strange as this is not a document ID. sent_sms_list does not contain number, only a state ? Does it follow the same order as list given as parameter ?